clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -4660,6 +4660,15 @@ void Clang::ConstructJob(Compilation &C, const JobAction &JA,
       CmdArgs.push_back("-fsycl-allow-func-ptr");
     }
 
+    // Forward -fsycl-instrument-device-code option to cc1. This option can only
+    // be used for SPIR-V-based targets.
+    if (Arg *A = Args.getLastArg(options::OPT_fsycl_instrument_device_code)) {
+      if (!Triple.isSPIR())
+        D.Diag(diag::err_drv_unsupported_opt_for_target)
+            << A->getAsString(Args) << TripleStr;
+      CmdArgs.push_back("-fsycl-instrument-device-code");
```

Comment:
Would it be better to only push `-fsycl-instrument-device-code` when we know it `isSPIR()`.  We can use `getLastArgNoClaim` and only claim when we use it, allowing for the diagnostic to be emitted as not being used for non-SPIR.

Reply:
> Would it be better to only push -fsycl-instrument-device-code when we know it isSPIR()?

My original intention, but somehow I've missed that my code does the opposite :)

>  We can use getLastArgNoClaim and only claim when we use it, allowing for the diagnostic to be emitted as not being used for non-SPIR.

Cool thought, will try this out! Basically, we'd allow such a compilation to happen anyway, but the diagnostic would be retained. I really don't think we're **obliged** to emit an error instead of a warning for such an option (the opposite might be better, actually).